### PR TITLE
Update psa.p4 with a separate clone() extern

### PR DESCRIPTION
**Pull request by hanw**

### CODE REVIEW

File: p4include/psa.p4 [hunk 1]
```diff
@@ -248,6 +223,36 @@ extern PacketReplicationEngine {
...
 
+extern Clone {
+  /// create a copy of the packet to the specified mirror session.
+  ///
+  /// The PSA specifies four types of cloning, with the packet sourced
+  /// from different points in the pipeline and sent back to ingress
+  /// or to the buffering queue in the egress (@see CloneMethod_t).
+  ///
+  /// @param clone_method  The type of cloning.
+  /// @param session_id    Port mirror session id.
```
Comment (cc10512) [inline p4include/psa.p4]: Parse error: "Port mirror session id". What do you mean by that?
And BTW, you need to also either pass a port or discuss how to set the port for the clone.

File: p4include/psa.p4 [hunk 2]
```diff
@@ -248,6 +223,36 @@ extern PacketReplicationEngine {
...
+  void clone (in CloneMethod_t clone_method, in SessionId_t session_id);
+
+  /// create a copy of the packet with additional data to the specified mirror session.
+  ///
+  /// The PSA specifies four types of cloning, with the packet sourced
+  /// from different points in the pipeline and sent back to ingress
+  /// or to the buffering queue in the egress (@see CloneMethod_t).
+  ///
+  /// @param clone_method  The type of cloning.
+  /// @param session_id    Port mirror session id.
```
Comment (cc10512) [inline p4include/psa.p4]: Same here.

File: p4include/psa.p4 [hunk 3]
```diff
@@ -248,6 +223,36 @@ extern PacketReplicationEngine {
   */
 }
 
+extern Clone {
+  /// create a copy of the packet to the specified mirror session.
```
Comment (cc10512) [inline p4include/psa.p4]: Need to capture the rules on how to define headers with leading session_id/clone_id and what to do in the parser. The example you had was good to clarify.

**Comment by hanw** [comment]
I believe this is no longer needed.

**Pull request closed without merging**